examples/python/guestbook/__main__.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2016-2018, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pulumi
+from pulumi_kubernetes.core.v1 import Service
+from pulumi_kubernetes.apps.v1 import Deployment
+
+redis_leader_labels = {
+    "app": "redis",
+    "tier": "backend",
+    "role": "master"
+}
+
+redis_leader_service = Service("redis-leader",
+    spec={
+        "ports": [{
+            "port": 6379,
+            "target_port": 6379,
+        }],
+        "selector": redis_leader_labels
+    },
+    metadata={
+        "name": "redis-master",
+        "labels": redis_leader_labels
+    })
+
+redis_leader_deployment = Deployment("redis-leader",
+    spec={
+        "selector": {
+            "match_labels": redis_leader_labels,
+        },
+        "replicas": 1,
+        "template": {
+            "metadata": {
+                "labels": redis_leader_labels,
+            },
+            "spec": {
+                "containers": [{
+                    "name": "master",
+                    "image": "k8s.gcr.io/redis:e2e",
+                    "resources": {
+                        "requests": {
+                            "cpu": "100m",
+                            "memory": "100Mi",
+                        },
+                    },
+                    "ports": [{
+                        "container_port": 6379,
+                    }],
+                }],
+            },
+        },
+    },
+    metadata={
+        "name": "redis-master"
+    })
+
+redis_follower_labels = {
+    "app": "redis",
+    "tier": "backend",
+    "role": "slave"
+}
+
+redis_follower_service = Service("redis-follower",
+    spec={
+        "ports": [{
+            "port": 6379,
+            "target_port": 6379,
+        }],
+        "selector": redis_follower_labels
+    },
+    metadata={
+        "name": "redis-slave",
+        "labels": redis_follower_labels 
+    })
+
+redis_follower_deployment = Deployment("redis-follower",
```

Comment:
Might want to rename this to `redis_slave_deployment` to match the naming in the example.

Reply:
I tried to explicitly avoid `slave` terminology whenever possible. Unfortunately it's not possible to do so everywhere here since it's hardcoded inside the docker image that we're deploying. If it's confusing, I'll stick to master/slave, but not without grimacing heavily.

Reply:
In point of fact, we have an org-wide policy of avoiding "slave" terminology. :) https://github.com/pulumi/examples/issues/133

Reply:
To clarify, my intent was to make the labels match, not to suggest that "slave" was a preferable term. The [upstream example](https://github.com/kubernetes/examples/tree/master/guestbook) still uses slave, and the mismatch jumped out at me. You could always change references to everything (except the image name) to "follower", and it wouldn't change the semantics of the example.

Reply:
I feel like I can't change the names of the services, since the PHP app in the frontend container is looking for them using DNS:

```
<b>Fatal error</b>:  Uncaught exception 'Predis\Connection\ConnectionException' with message 'php_network_getaddresses: getaddrinfo failed: Name or service not known [tcp://redis-slave:6379]' in /usr/local/lib/php/Predis/Connection/AbstractConnection.php:168
Stack trace:
#0 /usr/local/lib/php/Predis/Connection/StreamConnection.php(97): Predis\Connection\AbstractConnection-&gt;onConnectionError('php_network_get...', 0)
#1 /usr/local/lib/php/Predis/Connection/StreamConnection.php(58): Predis\Connection\StreamConnection-&gt;tcpStreamInitializer(Object(Predis\Connection\Parameters))
#2 /usr/local/lib/php/Predis/Connection/AbstractConnection.php(94): Predis\Connection\StreamConnection-&gt;createResource()
#3 /usr/local/lib/php/Predis/Connection/StreamConnection.php(158): Predis\Connection\AbstractConnection-&gt;connect()
#4 /usr/local/lib/php/Predis/Connection/AbstractConnection.php(193): Predis\Connection\StreamConnection-&gt;connect()
#5 /usr/local/lib/php/Predis/Connection/StreamConnection.php(184): Predis\Connection\AbstractConnection-&gt;getResour in <b>/usr/local/lib/php/Predis/Connection/AbstractConnection.php</b> on line <b>168</b><br />
```

Reply:
Ah, didn't occur to me that the app would have those names hardcoded. 🤷‍♂️